src/pages/namespace/Explorer/Tree/components/modals/CreateNew/Gateway/Consumer.tsx

```diff
@@ -0,0 +1,133 @@
+import {
+  DialogClose,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "~/design/Dialog";
+import { Network, PlusCircle } from "lucide-react";
+import { SubmitHandler, useForm } from "react-hook-form";
+
+import Button from "~/design/Button";
+import FormErrors from "~/componentsNext/FormErrors";
+import Input from "~/design/Input";
+import { addYamlFileExtension } from "../../../../utils";
+import { defaultConsumerFileYaml } from "~/pages/namespace/Explorer/Consumer/ConsumerEditor/utils";
+import { fileNameSchema } from "~/api/tree/schema/node";
+import { pages } from "~/util/router/pages";
+import { useCreateWorkflow } from "~/api/tree/mutate/createWorkflow";
+import { useNamespace } from "~/util/store/namespace";
+import { useNavigate } from "react-router-dom";
+import { useTranslation } from "react-i18next";
+import { z } from "zod";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+type FormInput = {
+  name: string;
+  fileContent: string;
+};
+
+const NewConsumer = ({
+  path,
+  close,
+  unallowedNames,
+}: {
+  path?: string;
+  close: () => void;
+  unallowedNames?: string[];
+}) => {
+  const { t } = useTranslation();
+  const namespace = useNamespace();
+  const navigate = useNavigate();
+  const {
+    register,
+    handleSubmit,
+    formState: { isDirty, errors, isValid, isSubmitted },
+  } = useForm<FormInput>({
+    resolver: zodResolver(
```

Comment:
This would be a bit nicer to read if we declared a `const resolver` separately, instead of nesting it here.